sympy/polys/polytools.py

```diff
@@ -6288,7 +6288,7 @@ def cancel(f, *gens, **args):
             raise PolynomialError(msg)
         # Handling of noncommutative and/or piecewise expressions
         if f.is_Add or f.is_Mul:
-            sifted = sift(f.args, lambda x: x.is_commutative and not x.has(Piecewise))
+            sifted = sift(f.args, lambda x: bool(x.is_commutative) and not x.has(Piecewise))
```

Comment:
If one of the operands is None, the expression returns None (instead of a Boolean). Quite strange indeed.

```
>>> a = None and True
>>> type(a)
<type 'NoneType'>
```